### PR TITLE
Add paypal payment search to admin page

### PR DESCRIPTION
**Pull request by sderickson**
Add a search form to the admin page for searching PayPal transactions, and add an index for those transactions.

### CODE REVIEW

File: server/middleware/payments.coffee [hunk 1]
```diff
@@ -5,11 +5,19 @@ parse = require '../commons/parse'
...
+  if req.query.payPalResource
+    query['$or'] = [
+      {'payPal.transactions.related_resources.sale.id': req.query.payPalResource},
+      {'payPalSale.id': req.query.payPalResource},
+    ]
+  dbq = Payment.find(query, parse.getProjectFromReq(req))
+  dbq.limit(parse.getLimitFromReq(req))
+  dbq.skip(parse.getSkipFromReq(req))
+  dbq.select(parse.getProjectFromReq(req))
+  payments = yield dbq.lean().exec()
```
Comment (differentmatt) [inline server/middleware/payments.coffee]: Why does this new version require an `exec`?

File: server/middleware/payments.coffee [hunk 2]
```diff
@@ -5,11 +5,19 @@ parse = require '../commons/parse'
...
+  query = {}
+  if req.query.nofree
+    query.amount = {$gt: 0}
+  if req.query.payPalResource
+    query['$or'] = [
+      {'payPal.transactions.related_resources.sale.id': req.query.payPalResource},
+      {'payPalSale.id': req.query.payPalResource},
+    ]
+  dbq = Payment.find(query, parse.getProjectFromReq(req))
+  dbq.limit(parse.getLimitFromReq(req))
```
Comment (differentmatt) [inline server/middleware/payments.coffee]: Will this limit break existing use cases that didn't have a limit applied?

File: server/middleware/payments.coffee (hunk 1 above)
Comment (sderickson) [reply server/middleware/payments.coffee]: I think I put it in there when debugging what turned out to be another thing. It's unnecessary, but not harmful, kind of like including `return` at the end of a CoffeeScript function or not.

File: server/middleware/payments.coffee (hunk 2 above)
Comment (sderickson) [reply server/middleware/payments.coffee]: Yes, will change this to not be limited by default.

**Comment by sderickson** [comment]
Confirmed the AnalyticsView continues to work, which is the only thing using this endpoint.

**Pull request merged**